skxray/roi.py

```diff
@@ -106,218 +104,122 @@ def rectangles(num_rois, roi_data, image_shape):
     return labels_grid
 
 
-def rings(image_shape, calibrated_center, num_rings,
-          first_r, delta_r):
+def rings(edges, center, shape):
     """
-    This will provide the indices of the required rings,
-    find the bin edges of the rings, and count the number
-    of pixels in each ring, and pixels indices for the
-    required rings when there is no step value between
-    rings.
+    Draw annual (ring-shaped) regions of interest.
+
+    Each ring will be labeled with an integer. Regions outside any ring will
+    be filled with zeros.
 
     Parameters
     ----------
-    image_shape: tuple
-        shape of the image (detector X and Y direction)
-        Order is (num_rows, num_columns)
-        shape is [image_shape[0], image_shape[1]])
+    edges: list
+        giving the inner and outer radius of each ring
+        e.g., [(1, 2), (11, 12), (21, 22)]
 
-    calibrated_center : tuple
-        defining the center of the image
-        (column value, row value) (mm)
-
-    num_rings : int
-        number of rings
+    center : rr, cc : tuple
 
-    first_r : float
-        radius of the first  ring
-
-    delta_r : float
-        thickness of the ring
+    shape: rr, cc: tuple
 
     Returns
     -------
-    labels_grid : array
-        indices of the required rings
-        shape is ([image_shape[0]*image_shape[1]], )
-
+    label_array : array
+        Elements not inside any ROI are zero; elements inside each
+        ROI are 1, 2, 3, corresponding to the order they are specified
+        in edges.
     """
-    grid_values = core.pixel_to_radius(image_shape, calibrated_center)
-
-    edges = rings_edges(num_rings, first_r, delta_r)
-
-    # indices of rings
-    labels_grid = np.digitize(np.ravel(grid_values), np.array(edges),
+    edges = np.asarray(edges)
```

Comment:
I would throw this through a `np.atleast_2d` call which will up convert `[1, 2]` correctly.